### ISSUE TITLE
Question about decoder loss

### ISSUE THREAD

**Issue by yiyulics** [Neur-IO/ReVQ#4]
Hi,

I have noticed that in the following loss functions, the `loss_decoder` only updates the decoder but does not update the quantizer, because a `.detach()` operation is applied before entering the decoder.

https://github.com/Neur-IO/ReVQ/blob/0d6634d823fd64ad7d0a3315382356948d08dfd0/scripts/train.py#L133-L139

May I ask why you design it this way, rather than using the decoder loss to update the decoder and quantizer altogether (i.e., without the `.detach()` operation before entering the decoder)?
